packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -639,11 +642,47 @@ export function makeVirtualObjectManager(
   }
 
   function defineKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, false);
+    const kindID = `${allocateExportID()}`;
+    return defineKindInternal(kindID, tag, init, actualize, finish, false);
+  }
+
+  let kindIDID;
+  const kindDescriptors = new WeakMap();
+
+  function reanimateDurableKindID(vobjID, _proforma) {
+    const { subid: kindID } = parseVatSlot(vobjID);
+    const raw = syscall.vatstoreGet(`vom.kind.${kindID}`);
+    assert(raw, X`unknown kind ID ${kindID}`);
+    const durableKindDescriptor = JSON.parse(raw);
+    const kindHandle = Far('kind', {});
+    kindDescriptors.set(kindHandle, durableKindDescriptor);
+    return kindHandle;
   }
 
-  function defineDurableKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, true);
+  const makeKindHandle = tag => {
+    if (!kindIDID) {
+      kindIDID = `${allocateExportID()}`;
+      syscall.vatstoreSet('kindIDID', kindIDID);
+      vrm.registerKind(kindIDID, reanimateDurableKindID, () => null, true);
+    }
+    const kindID = `${allocateExportID()}`;
+    const kindIDvref = `o+${kindIDID}/${kindID}`;
+    const durableKindDescriptor = { kindID, tag };
+    const kindHandle = Far('kind', {});
+    kindDescriptors.set(kindHandle, durableKindDescriptor);
+    registerValue(kindIDvref, kindHandle, false);
```

Comment:
I'm a little bit worried that `registerValue` won't be sufficient; that we should really use some form of `makeKindHandle = defineDurableKind(..)` to define these things, so that e.g. serialize/unserialize and GC and export-status and everything will behave exactly the same (well-tested) way that normal kinds do. This would require something sneaky to break the circular reference, of course. `defineDurableKind` only needs the `kindHandle` to get a descriptor, so what if we broke `defineDurableKind` into an internal form (which takes the descriptor directly), and an external form (which takes a `kindHandle`).

Then we could build `durableKindHandleDescriptor = { kindID: kindIDID, tag: 'kind' }` and then `makeKindHandle = defineDurableKindInternal(durableKindHandleDescriptor, ( kindHandle behavior ...))`.

Reply:
I guess the lack of state and multiple facets on a `kindHandle` means that we wouldn't be using a lot of the functionality of `define(Durable)Kind`. I'm just worried about us making some change to e.g. the way `unserialize` works, applying that change to `defineKindInternal` or `makeRepresentative`, but forgetting to apply it to `reanimateDurableKindID`.

Reply:
> so what if we broke defineDurableKind into an internal form (which takes the descriptor directly), and an external form (which takes a kindHandle).

I'm confused about what you're proposing here, since what you describe is how it works now.